src/sagemaker/estimator.py

```diff
@@ -660,7 +667,52 @@ def __init__(
         self.checkpoint_local_path = checkpoint_local_path
 
         self.rules = rules
-        self.debugger_hook_config = debugger_hook_config
+
+        # First get the config value provided for debugger_hook_config
+        # If debugger_hook_config is provided as a dict in the sagemaker config,
+        # then resolve it from the config,
+        # else resolve in the priority sequence as 1. the provided value,
+        # 2. boolean value from config
+        inferred_debugger_hook_config = get_sagemaker_config_value(
+            self.sagemaker_session, TRAINING_JOB_DEBUG_HOOK_CONFIG_PATH
+        )
+        if isinstance(inferred_debugger_hook_config, dict):
+            self.debugger_hook_config = resolve_class_attribute_from_config(
```

Comment:
nit: one of the other helpers that merges in a whole dict might work instead of doing these one by one. Tho not sure how it would behave with the property lists